pkg/cli/mirror/mirror.go

```diff
@@ -556,13 +557,15 @@ func (o *MirrorOptions) mirrorImages(ctx context.Context, cleanup cleanupFunc) e
 	return cleanup()
 }
 func (o *MirrorOptions) mirrorOCIImages(cleanup cleanupFunc) error {
-	remoteRegFuncs := RemoteRegFuncs{
-		pull:           crane.Pull,
-		saveOCI:        crane.SaveOCI,
-		saveLegacy:     crane.SaveLegacy,
-		push:           copy.Image,
-		load:           crane.Load,
+	o.remoteRegFuncs = RemoteRegFuncs{
```

Comment:
Can we add the context to the whole `mirrorOCIImages` function pathway? It's available via `cmd.Context()` and it could be passed along where ever it is needed instead of using `context.Background()` or `context.TODO()`.

Reply:
Hi @jchunkins 
Sorry, I don't think I get what you mean. 
The context parameters in [L563](https://github.com/openshift/oc-mirror/blob/e00c92d43150dd37c25c7ccdff0e7fe037fd8057/pkg/cli/mirror/mirror.go#L563) and [L566](https://github.com/openshift/oc-mirror/blob/e00c92d43150dd37c25c7ccdff0e7fe037fd8057/pkg/cli/mirror/mirror.go#L566) are from the method signatures of [ImageReference.NewImageSource](https://pkg.go.dev/github.com/containers/image/v5@v5.22.0/types#ImageReference.NewImageSource)
 and [ImageSource.GetManifest](https://pkg.go.dev/github.com/containers/image/v5@v5.22.0/types#ImageSource.GetManifest)

Reply:
Sorry, let me explain what I mean... The function `mirrorOCIImages` is currently the "entry point" for the OCI features. It's being called in `func (o *MirrorOptions) Run(cmd *cobra.Command, f kcmdutil.Factory) (err error) {` in mirror.go. That `Run` function contains the cobra command argument `cmd`, which allows you to call `cmd.Context()` to get the `context.Context` instance for the CLI. You should pass that context object into the `mirrorOCIImages` function so that it can in turn pass it on to `bulkImageCopy` and `bulkImageMirror`. There are many locations in the copy.go that are using `context.TODO()` or `context.Background()` that should instead be using the context provided by the CLI.

Reply:
Done in 3614339574c23025f373b547e9e3e03dd9d5d6b7